src/main/java/com/redhat/devtools/lsp4ij/client/features/EditorBehaviorFeature.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.client.features;
+
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LanguageServiceAccessor;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Client-side editor behavior feature. This does not correspond to an actual LSP feature.
+ */
+@ApiStatus.Experimental
+public class EditorBehaviorFeature {
+
+    private LSPClientFeatures clientFeatures;
+
+    /**
+     * Creates the editor behavior feature for the containing client features.
+     *
+     * @param clientFeatures the client features
+     */
+    public EditorBehaviorFeature(@NotNull LSPClientFeatures clientFeatures) {
+        this.clientFeatures = clientFeatures;
+    }
+
+    /**
+     * Sets the containing client features.
+     *
+     * @param clientFeatures the client features.
+     */
+    void setClientFeatures(@NotNull LSPClientFeatures clientFeatures) {
+        this.clientFeatures = clientFeatures;
+    }
+
+    /**
+     * Returns the containing client features.
+     *
+     * @return the client features
+     */
+    @NotNull
+    protected LSPClientFeatures getClientFeatures() {
+        return clientFeatures;
+    }
+
+    /**
+     * Whether or not editor improvements for string literals are enabled. Defaults to false.
+     *
+     * @param file the file
+     * @return true if editor improvements for string literals are enabled; otherwise false
+     */
+    public boolean isEnableStringLiteralImprovements(@NotNull PsiFile file) {
+        // Default to disabled
+        return false;
```

Comment:
These are all _disabled by default_ here, but are _enabled by default_ in `UserDefinedEditorBehaviorFeature` where they can be easily disabled via client config.